src/main/java/domain/Ladder.java

```diff
@@ -16,38 +14,50 @@ public class Ladder {
 
     public Ladder(final Width width, final Height height, final ScaffoldGenerator scaffoldGenerator) {
         this.lines = IntStream.range(0, height.getValue())
-                .mapToObj(it -> createLine(width, scaffoldGenerator))
+                .mapToObj(it -> new Line(width, scaffoldGenerator))
                 .collect(Collectors.toUnmodifiableList());
         validateLines(lines);
     }
 
-    private Line createLine(final Width width, final ScaffoldGenerator scaffoldGenerator) {
-        return new Line(width, scaffoldGenerator);
-    }
-
-    private static void validateLines(final List<Line> lines) {
+    private void validateLines(final List<Line> lines) {
         validateLineSizeEmpty(lines);
         validateLinesSameSize(lines);
     }
 
-    private static void validateLineSizeEmpty(final List<Line> lines) {
+    private void validateLineSizeEmpty(final List<Line> lines) {
         if (lines.isEmpty()) {
             throw new IllegalArgumentException(ERROR_LINE_SIZE_EMPTY);
         }
     }
 
-    private static void validateLinesSameSize(final List<Line> lines) {
+    private void validateLinesSameSize(final List<Line> lines) {
         for (int i = 0; i < lines.size() - 1; i++) {
```

Comment:
for 반복문 안에서 size() 와 같이 메서드를 호출하면, 반복문을 돌 때마다 메서드 호출이 발생할 수 있어요!

미리 변수로 할당하여 사용하는 방법이 있습니다 😸 
또는 다른 곳에서 사용하신것처럼 IntStream을 사용하는 방법도 있겠네요
```suggestion
        int bound = lines.size() - 1;
        for (int i = 0; i < bound; i++) {
        
        // 또는
        IntStream.range(0, lines.size() - 1)
                .forEach(i -> validateLineSameSize(lines, i));
```

Reply:
사실 이전에 IntStream을 사용했던 이유는 요구사항중 하나인 depth 만족이었습니다

우테코 본과정을 시작하며 처음 Stream을 사용하기 시작했고,
'손에 익혀보자 익혀보자'를 목표로,  stream을 최대한 적용하려 노력하고있습니다.
동시에 무분별한 stream 사용을 지양해야 한다는 리뷰 및 포스트도 많이 접하고 있는데요,
[Stream.forEach()를 지양하라는 글도 보고있구요]https://tecoble.techcourse.co.kr/post/2020-05-14-foreach-vs-forloop/
아직 Stream을 써야할 때와 쓰지 말아야할 때에 대한 주관이 잡히지 않았습니다.

찰리는 어떨 때 stream을 사용하고, 어떨 때 for-loop를 사용하나요?

찰리의 stream, for-loop 를 선택하는 기준이 궁금합니다!

Reply:
좋은 고민이네요 👍 

> 어떨 때 stream을 사용하고, 어떨 때 for-loop를 사용하나요?

Stream 의 장점은 무엇인지 생각해보면 될것같아요!
장점으로 제일 먼저 생각나는건 가독성이고, 추가적으로 병렬 실행 가능 등이 있겠군요 😄 

그렇기에 제 선택기준은 `for-loop vs stream 누가 더 가독성이 좋을까?` 를 비교해보는 것입니다.
- steam의 flatmap 같은것을 써봤더니 나도 이해하기 어려운 코드가 된것같아 -> 중첩 for 문으로 해결!
- 어떤 팀은 Stream 이 익숙하지 않아서 for-loop가 더 읽기 좋아요~ 라고하면 그것을 컨벤션으로 사용할 수도 있을것입니다.

이런글도 참고해보시면 `좋겠네요!`
[Collection.forEach와 Stream.forEach는 뭐가 다를까?](https://tecoble.techcourse.co.kr/post/2020-09-30-collection-stream-for-each/)